src/components/Goals/BikeGoal/DateSelector/helpers.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+
+// TODO: remove eslint-disable by convert 'src/lib/timeseries' in TS
+// or at least getEarliestTimeserie
+
+import type { TimeSeries } from 'cozy-client/types/models/timeseries'
+
+import differenceInCalendarYears from 'date-fns/differenceInCalendarYears'
+import subYears from 'date-fns/subYears'
+
+import { getEarliestTimeserie } from 'src/lib/timeseries'
+
+export const makeMenuDates = (timeseries: Array<TimeSeries>): Array<string> => {
+  const latestDate = new Date()
+  if (!timeseries || timeseries.length === 0) {
```

Comment:
Assuming a strict config and that the call site of the function is also written in TypeScript (which is not the case currently), `!timeseries || ` should not be needed because here `timeseries` can't be falsy since this is an array.
But i agree this is safer right now.